Fix torch.library.register_fake's module reporting

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125037

torch.library.register_fake reports the python module the fake impl is
located in. This is used to check against
`m.set_python_module("foo.bar")` calls in C++.

The module reporting logic was wrong in most cases. This PR fixes it.

Test Plan:
- exhaustive tests

zou3519 committed:
Fix torch.library.register_fake's module reporting

torch.library.register_fake reports the python module the fake impl is
located in. This is used to check against
`m.set_python_module("foo.bar")` calls in C++.

The module reporting logic was wrong in most cases. This PR fixes it.

Test Plan:
- exhaustive tests

[ghstack-poisoned]

albanD commented on torch/testing/_internal/custom_op_db.py:
```diff
@@ -435,3 +435,54 @@ def sample_inputs_numpy_nms(opinfo, device, dtype, requires_grad, **kwargs):
+# some mechanical test cases
+# ==============================================================
+
+lib = torch.library.Library("_torch_testing", "FRAGMENT")  # noqa: TOR901
```
Wait this would always get registered if anyone end up importing this by mistake with torch. ?

zou3519 replied:
Yes. All the custom ops in this file get registered if they get imported. It's okay because they're in a namespace that users shouldn't use.

zou3519 committed:
Update on "Fix torch.library.register_fake's module reporting"


torch.library.register_fake reports the python module the fake impl is
located in. This is used to check against
`m.set_python_module("foo.bar")` calls in C++.

The module reporting logic was wrong in most cases. This PR fixes it.

Test Plan:
- exhaustive tests

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge
